diffprivlib/models/tree.py

```diff
@@ -0,0 +1,602 @@
+            X : array-like, shape (n_samples, n_features)
+                Training vector, where n_samples is the number of samples and n_features is the number of features.
+
+        Raises:
+            Exception: If the model is not fitted before prediction.
+
+        Returns:
+            [nd.array]: Numpy array of predictions.
+        """
+        if not self.check_is_fitted():
+            raise Exception('Model is not fitted yet')
+
+        preds = []
+
+        for estimator in self._estimators:
+            preds.append(np.array(estimator.predict(X)).reshape(-1, 1))
+
+        y = np.hstack(preds)
+
+        return np.apply_along_axis(lambda x: Counter(list(x)).most_common(1)[0][0], arr=y, axis=1)
+
+
+class DecisionTreeClassifier(ClassifierMixin, BaseDecisionTree):
+    r"""Decision Tree Classifier with differential privacy.
+
+    This class implements the base differentially private decision tree classifier
+    for the Random Forest classifier algorithm. Not meant to be used separately.
+
+    Parameters
+    ----------
+    epsilon: float, default: 1.0
+        Privacy parameter :math:`\epsilon`.
+
+    cat_feature_threshold: int, default: 10
+        Threshold value used to determine categorical features. For example, value of ``10`` means
+        any feature that has less than or equal to 10 unique values will be treated as a categorical feature.
+
+    max_depth: int, default: 15
+        The maximum depth of the tree.
+
+    random_state: float, optional
+        Sets the numpy random seed.
+
+    feature_domains: dict, optional
+        A dictionary of domain values for all features where keys are the feature indexes in the training data and
+        the values are an array of domain values for categorical features and an array of min and max values for
+        continuous features. For example, if the training data is [[2, 'dog'], [5, 'cat'], [7, 'dog']], then
+        the feature_domains would be {'0': [2, 7], '1': ['dog', 'cat']}. If not provided, feature domains will
+        be constructed from the data, but this will result in :class:`.PrivacyLeakWarning`.
+    """
+    def __init__(self, cat_feature_threshold=10, max_depth=15, epsilon=1, random_state=None, feature_domains=None):
+        super().__init__(
+            criterion=None,
+            splitter=None,
+            max_depth=max_depth,
+            min_samples_split=None,
+            min_samples_leaf=None,
+            min_weight_fraction_leaf=None,
+            max_features=None,
+            random_state=random_state,
+            max_leaf_nodes=None,
+            min_impurity_decrease=None,
+            min_impurity_split=None
+        )
+        self._feature_domains = feature_domains
+        self._cat_features = None
+        self._classes = None
+        self._cat_feature_threshold = cat_feature_threshold
+        self._max_depth = max_depth
+        self._epsilon = epsilon
+        self._root = None
+        self._fitted = False
```

Comment:
Same as for the main class, parameters shouldn't have a leading underscore, fitted attributes shouldn't be defined here, and should have a trailing underscore.